messaging/src/main/java/org/axonframework/common/infra/StringBuilderComponentDescriptor.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (c) 2010-2025. Axon Framework
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.common.infra;
+
+import jakarta.annotation.Nonnull;
+
+import java.util.Collection;
+import java.util.Map;
+
+/**
+ * A {@link ComponentDescriptor} implementation that uses a {@link StringBuilder} to construct the result.
+ *
+ * @author Steven van Beelen
+ * @since 5.0.0
+ */
+public class StringBuilderComponentDescriptor implements ComponentDescriptor {
```

Comment:
I'm not sure if it makes sense to do it with some "StringBuilder", because we don't define format here. For me it looks like Json implementation, but without any library.

Reply:
I am entirely fine with throwing this away :+1:

Reply:
Yes, it should be fairly easy, but I don't think it's good to provide that. If you have "custom version", you need to define some custom format and if we'd like to use it somehwere, like console we also need a "custom" parser. So i recommend adhering to some well-known formats. 

Jackson it's indeed an additional dependency, but it's already present in the framework.